src/builtin.c

```diff
@@ -548,6 +564,38 @@ static jv f_format(jq_state *jq, jv input, jv fmt) {
     }
     jv_free(input);
     return line;
+  } else if (!strcmp(fmt_s, "base64d")) {
+    jv_free(fmt);
+    input = f_tostring(jq, input);
+    const unsigned char* data = (const unsigned char*)jv_string_value(input);
+    int len = jv_string_length_bytes(jv_copy(input));
+    char result[(3 * len) / 4]; // decoded result buffer
```

Comment:
VLAs are dangerous.  Suppose the incoming string is 10MB... and suppose the stack grows down...  you could easily start by writing to memory that's not on the stack but in some other memory mapping.  In the best case you'd just segfault. I would much prefer this be malloc()ed.